bson/bsoncodec/uint_codec.go

```diff
@@ -164,11 +164,15 @@ func (uic *UIntCodec) decodeType(dc DecodeContext, vr bsonrw.ValueReader, t refl
 
 		return reflect.ValueOf(uint64(i64)), nil
 	case reflect.Uint:
-		if i64 < 0 || int64(uint(i64)) != i64 { // Can we fit this inside of an uint
+		if i64 < 0 {
+			return emptyValue, fmt.Errorf("%d overflows uint", i64)
```

Comment:
Use a more accurate error message.

```suggestion
			return emptyValue, fmt.Errorf("%d underflows uint", i64)
```

Reply:
It seems better to keep the error message consistent with other unsigned types.

Reply:
Sounds reasonable. We can fix them all at the same time. Consider this comment resolved.